server/mldb_server.cc

```diff
@@ -474,20 +495,36 @@ scanPlugins(const std::string & dir_)
                 return true;
             };
 
-        try {
+        errorWrapper([&]() {
             forEachUriObject(dir, onFile, onSubdir);
-        } catch (const HttpReturnException & exc) {
-            logger->error() << "error scanning plugin directory "
-                            << dir << ": " << exc.what();
-            logger->error() << "details:";
-            logger->error() << jsonEncode(exc.details);
-            logger->error() << "plugins will be ignored";
-            return;
-        } catch (const std::exception & exc) {
-            logger->error() << "error scanning plugin directory  "
-                            << dir << ": " << exc.what();
-            logger->error() << "plugins will be ignored";
-            return;
+        });
+    }
+
+    // Dumb algo, loop until nothing is left or the last pass didn't load anything
```

Comment:
We resolve the dependencies with a dumb loop. There should not be a lot so perf is not an issue.